torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -930,6 +931,36 @@ bool AliasDb::couldMoveBeforeTopologically(Node* n, Node* movePoint) {
   return tryMove(n, movePoint, MoveSide::BEFORE, /*dryRun=*/true);
 }
 
+bool AliasDb::hasWriters(const at::ArrayRef<Value*>& values) const {
+  return std::any_of(values.begin(), values.end(), [&](Value* value) {
+    return hasWriters(value);
+  });
+}
+
+// Correctness conditions:
+// no values in either set can have writers, and values in both sets
+// cannot escape the current graph scope. Values can escape the current scope
+// by aliasing a graph output or input, or aliasing an input to
+// prim::FunctionCall or MethodCall. FunctionCall and MethodCall are marked as
+// mutating ops, so we do not need to check for them because they are covered
+// by the writers condition.
+bool AliasDb::safeToChangeAliasingRelationship(
+    const at::ArrayRef<Value*>& a,
+    const at::ArrayRef<Value*>& b) const {
+  if (hasWriters(a) || hasWriters(b)) {
+    return false;
+  }
+  auto graph_outputs = graph_->outputs();
+  auto graph_inputs = graph_->inputs();
+
+  bool a_escapes_scope =
+      mayContainAlias(graph_inputs, a) || mayContainAlias(graph_outputs, b);
+  bool b_escapes_scope =
+      mayContainAlias(graph_inputs, a) || mayContainAlias(graph_outputs, b);
```

Comment:
both these bools compute the same thing, seems like a copy/paste error. But if it doesn't fail any tests, it also indicates this function is pretty undertested.